internal/adapter/outbound/mongodb/user_repository.go

```diff
@@ -0,0 +1,148 @@
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/Kercyn/crud_template/configs"
+	"github.com/Kercyn/crud_template/internal/adapter/outbound/mongodb/dto"
+	"github.com/Kercyn/crud_template/internal/core/domain"
+	port "github.com/Kercyn/crud_template/internal/core/port/outbound"
+	apperror "github.com/Kercyn/crud_template/pkg/error"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type MongoUserRepository struct {
+	ctx    context.Context
```

Comment:
What is the purpose of `ctx` in this struct?

Reply:
It allows us to inject information such as timeouts for our database calls, as seen on [internal/app/root.go](https://github.com/GlobalWebIndex/platform-go-challenge/pull/11/files#diff-e13d4e4bf77c274abdef9106db2aa3792527330a3d3a3ddb8f35a84c139a1822), line 17.